packages/commons/src/userAgentMiddleware.ts

```diff
@@ -33,6 +33,13 @@ const customUserAgentMiddleware = (feature: string) => {
 // @ts-ignore
```

Comment:
not sure I like ignoring things in the type system

Reply:
Acknowledged and agree.

In this instance we decided to use an `any` as a type for the `client` because we didn't want to bring in the `@aws-sdk/types` package into the utility. This code resides in the `commons` utility, which is pulled in by all the other Powertools utilities.

Given that there's all the core utilities don't use any AWS SDK, using the `@aws-sdk/types` in this module would require customers to install it as a developer dependency when they develop or build their code using Powertools.
This is a problem because when building with TypeScript, the type-checking engine will block the compilation is the dependency is missing.

To avoid this, we decided to go for an `any` type (which is discouraged in our codebase - hence the `@ts-ignore`) and also put [unit tests](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/commons/tests/unit/userAgentMiddleware.test.ts#L89-L114).

We have considered writing our own minimal `Client` type, and the option is still in the cards, but have decided to not include it for now given the limited scope of the feature (i.e. getting a client from a 3rd party library and calling a method on it) and the fact that we implemented the function in a way that even if you pass a random object (aka not a client) it won't throw an error but just log a warning.

I am still strongly inclined to review this type as soon as we have bandwidth and remove the ignore.

Reply:
I think my preference would be to write our own types def.